File: 674_Longest Continuous Increasing Subsequence.java
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        int res = 0, tmpLen = 1, tmpMax = nums[0];

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > tmpMax) tmpLen++;
            else {
                res = Math.max(res, tmpLen);
                tmpLen = 1;
            }
            tmpMax = nums[i];
        }
        return Math.max(res, tmpLen);
    }
}

File: 674_Longest Continuous Increasing Subsequence.py
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        if len(nums) <= 1: return len(nums)
        l = res = 0
        for r in range(1, len(nums)):
            if nums[r] <= nums[r-1]: l = r
            res = max(res, r - l + 1)
        return res